cmd/mode.go

```diff
@@ -119,17 +123,8 @@ func (r *Mode) getArtifact() (err error) {
 	return
 }
 
-//
 // mavenArtifact get maven artifact.
-func (r *Mode) mavenArtifact(application *api.Application) (err error) {
-	binDir := path.Join(BinDir, "maven")
-	maven := repository.Maven{
-		M2Dir:  M2Dir,
-		BinDir: binDir,
-		Remote: repository.Remote{
-			Identities: application.Identities,
-		},
-	}
+func (r *Mode) mavenArtifact(application *api.Application, maven *repository.Maven, binDir string) (err error) {
```

Comment:
Can't the code below use maven.BinDir instead of passing separately?

Reply:
Updated that is much cleaner!